### PR TITLE
Run worker binary

### PR DESCRIPTION
**Pull request by rtannenbaum**
Tested running the worker with the Google cloud extension, with our build/run docker script (in config/gcp/)

Currently the Google cloud extension and Microsoft transfer extension are hardcoded since service loading isn't working yet. I left some TODOs for @seehamrun on service loading the extensions.

Left a TODO for flag parsing as well, currently the worker's ExtensionContext is hardcoded to return local config values

#172

### CODE REVIEW

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java [hunk 1]
```diff
@@ -22,24 +22,27 @@
...
 import java.lang.annotation.Target;
 import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.spi.cloud.extension.CloudExtensionModule;
+import org.dataportabilityproject.spi.cloud.storage.BucketStore;
+import org.dataportabilityproject.spi.cloud.storage.CryptoKeyStore;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
 
 /** Bindings for cloud platform components using Google Cloud Platform. * */
-final class GoogleCloudModule extends AbstractModule {
+final class GoogleCloudModule extends CloudExtensionModule {
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java]: Now or in the next PR, can we name it GoogleCloudExtensionModule for consistency?

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java [hunk 2]
```diff
@@ -52,6 +55,12 @@
   // The path where Kubernetes stores Secrets.
   private static final String KUBERNETES_SECRETS_PATH_ROOT = "/var/secrets/";
 
+  private final ExtensionContext context;
+
+  GoogleCloudModule(ExtensionContext context) {
+    this.context = context;
```
Comment (holachuy) [inline extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java]: Maybe as this evolves, we can pass the objects provided by the context to another Constructor so we can test directly

GoogleCloudModule(HttpTransport transport, JSONFactor factory) {
 this.transport = transport;
 this.factory = factory;
}

File: libraries/security/src/main/java/org/dataportabilityproject/security/RsaSymmetricKeyGenerator.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 import org.slf4j.LoggerFactory;
 
 /** RSA-based implementation for {@link KeyPair} creation and encoding. */
-class RsaSymmetricKeyGenerator implements AsymmetricKeyGenerator {
+public class RsaSymmetricKeyGenerator implements AsymmetricKeyGenerator {
```
Comment (holachuy) [inline libraries/security/src/main/java/org/dataportabilityproject/security/RsaSymmetricKeyGenerator.java]: TODO, provide a public module to bind KeyGenerators

File: portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/extension/CloudExtensionModule.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.google.inject.AbstractModule;
+import org.dataportabilityproject.spi.cloud.storage.BucketStore;
+import org.dataportabilityproject.spi.cloud.storage.CryptoKeyStore;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+
+/**
+ * Guice module that must be implemented by Cloud extensions.
+ */
+public class CloudExtensionModule extends AbstractModule {
```
Comment (holachuy) [inline portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/extension/CloudExtensionModule.java]: Do we expect subclasses to call super.configure() in their configure() impl?

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  private static final JsonFactory JSON_FACTORY = new JacksonFactory();
+
+  @Override
+  public Logger getLogger() {
+    return null;
+  }
+
+  @Override
+  public <T> T getService(Class<T> type) {
+    if (type.equals(HttpTransport.class)) {
```
Comment (holachuy) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: We can also put objects in a map keyed by type and then just read from the map here. In the future as the list gets longer.

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      return (T) JSON_FACTORY;
+    }
+    return null;
+  }
+
+  @Override
+  public <T> T getConfiguration(String key, T defaultValue) {
+    // TODO parse flags. Returning hardcoded local+GOOGLE config now for testing.
+    if (key.equals("cloud")) {
+      return (T) "GOOGLE";
```
Comment (holachuy) [inline portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: Constant

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java (hunk 1 above)
Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java]: good point, done

File: extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java (hunk 2 above)
Comment (rtannenbaum) [reply extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudModule.java]: I had it that way first actually, I didn't want the params list to get too long as it's manually called, I'm fine with either way though. I'll leave for now and we can always change it later

File: portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/extension/CloudExtensionModule.java (hunk 4 above)
Comment (rtannenbaum) [reply portability-spi-cloud/src/main/java/org/dataportabilityproject/spi/cloud/extension/CloudExtensionModule.java]: whoops, yes, good catch

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java (hunk 5 above)
Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: Done

File: portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java (hunk 6 above)
Comment (rtannenbaum) [reply portability-worker/src/main/java/org/dataportabilityproject/worker/WorkerExtensionContext.java]: it's temporarily hardcoded til we parse flags so I'll address this then

File: libraries/security/src/main/java/org/dataportabilityproject/security/RsaSymmetricKeyGenerator.java (hunk 3 above)
Comment (rtannenbaum) [reply libraries/security/src/main/java/org/dataportabilityproject/security/RsaSymmetricKeyGenerator.java]: Done

**Comment by holachuy** [comment]
I'll go ahead and submit it since it's late.

**Pull request merged**